mscore/editstaff.cpp

```diff
@@ -61,6 +60,61 @@ EditStaff::EditStaff(Staff* s, int /*tick*/, QWidget* parent)
       minPitchPSelect->setIcon(editIcon);
       maxPitchPSelect->setIcon(editIcon);
 
+      staff = nullptr;
+      setStaff(s);
+
+      if (!useFactorySettings) {
+            QSettings settings;
+            settings.beginGroup("EditStaff");
+            resize(settings.value("size", QSize(484, 184)).toSize());
+            move(settings.value("pos", QPoint(10, 10)).toPoint());
+            settings.endGroup();
+            }
+
+      connect(buttonBox,            SIGNAL(clicked(QAbstractButton*)), SLOT(bboxClicked(QAbstractButton*)));
+      connect(changeInstrument,     SIGNAL(clicked()),            SLOT(showInstrumentDialog()));
+      connect(changeStaffType,      SIGNAL(clicked()),            SLOT(showStaffTypeDialog()));
+      connect(minPitchASelect,      SIGNAL(clicked()),            SLOT(minPitchAClicked()));
+      connect(maxPitchASelect,      SIGNAL(clicked()),            SLOT(maxPitchAClicked()));
+      connect(minPitchPSelect,      SIGNAL(clicked()),            SLOT(minPitchPClicked()));
+      connect(maxPitchPSelect,      SIGNAL(clicked()),            SLOT(maxPitchPClicked()));
+      connect(editStringData,       SIGNAL(clicked()),            SLOT(editStringDataClicked()));
+      connect(lines,                SIGNAL(valueChanged(int)),    SLOT(numOfLinesChanged()));
+      connect(lineDistance,         SIGNAL(valueChanged(double)), SLOT(lineDistanceChanged()));
+      connect(showClef,             SIGNAL(clicked()),            SLOT(showClefChanged()));
+      connect(showTimesig,          SIGNAL(clicked()),            SLOT(showTimeSigChanged()));
+      connect(showBarlines,         SIGNAL(clicked()),            SLOT(showBarlinesChanged()));
+
+      connect(nextButton,           SIGNAL(clicked()),            SLOT(gotoNextStaff()));
+      connect(previousButton,       SIGNAL(clicked()),            SLOT(gotoPreviousStaff()));
+
+      addAction(getAction("local-help"));  // why is this needed?
+      }
+
+//---------------------------------------------------------
+//   closeEvent
+//---------------------------------------------------------
+
+void EditStaff::closeEvent(QCloseEvent* ev)
+      {
+      QSettings settings;
+      settings.beginGroup("EditStaff");
+      settings.setValue("size", size());
+      settings.setValue("pos", pos());
+      settings.endGroup();
+      QWidget::closeEvent(ev);
+      }
+
+//---------------------------------------------------------
+//   setStaff
+//---------------------------------------------------------
+
+void EditStaff::setStaff(Staff* s)
+      {
+      if (staff != nullptr)
+            delete staff;
+
+      orgStaff = s;
       Part* part        = orgStaff->part();
       instrument        = *part->instrument(/*tick*/);
       Score* score      = part->score();
```

Comment:
It won't let me comment on the non-changed lines below, so here goes:
I'm wondering whether the `staff->set…` calls on these lines couldn't be changed into `staff->init(orgStaff)`. IMHO it would improve the readability of this part.

The init function does the following extra:
```
      setDefaultClefType(s->defaultClefType());
      _brackets          = s->_brackets;
      _barLineSpan       = s->_barLineSpan;
      _barLineFrom       = s->_barLineFrom;
      _barLineTo         = s->_barLineTo;
```
of which I believe only the `setDefaultClefType` might be an issue, as this is a pass-by-reference call as far as I can see